site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,196 @@
+
+\[2\] it is in theory possible to write global position deletes, but it is always preferred to write partition position deletes because the writer already knows the exact partition to use after the scan.
+
+\[3\] if the input inequality predicate cannot be easily converted to a finite number of equality predicates (e.g. `price > 2.33`), then it is preferred to use position deletes instead.
+
+For example, in the change data capture use case, both partition position deletes and partition equality deletes are used. 
+For a new delete such as `id = 1001`, the writer checks the unflushed data index to see if there is an existing row matching `id = 1001` and performs a position delete.
+It also writes an equality delete that is applied to all the existing data files that are already in the table.
+
+### Reading data with delete files
+
+During Iceberg's scan file planning phase, a delete file index is constructed to filter the delete files associated with any data file, with the following rules:
+
+1. equality delete files are applied to data files of strictly lower sequence numbers
+2. position delete files are applied to data files of equal or lower sequence numbers
+3. further pruning is performed by comparing the partition and statistics information of each pair of delete and data file. Therefore partition deletes are always preferred to global deletes.
+
+After the planning phase, each data file to read is associated with a set of delete files to merge.
+Because position deletes must be sorted by file path and row position based on the spec,
+applying position deletes to data files can be viewed as merging two sorted lists, which can be done efficiently.
+When a position delete file is too large, we can also stream the rows without the need to buffer the entire file into memory.
+In contrast, all the rows of all the associated equality delete files must be loaded to memory to be checked against every row in a data file.
+Therefore, having too many equality deletes not only is computationally less efficient but also runs the risk of out-of-memory error.
+
+#### Position delete vs equality delete
+
+From the performance perspective, position delete is preferred based on its read mechanism. However, the advantage in read has the following tradeoffs:
+
+1. slower write due to scanning
+2. potential commit conflicts with operations that tries to remove data files
+
+The slower write of position delete is already explained the [writing delete files](#writing-delete-files) section.
+Here we use an example to show the issue of potential commit conflicts.
+Consider the following case, where a `rewrite` starts at `t1`, a `delete` starts at `t2` before `rewrite` completes.
+These two operations are running against the same table snapshot of sequence number `100`:
+
+```text
+t0: data.1 data.2 data.3 delete.3_1 (seq=100)
+       \     /       \     /
+t1:     data.4        data.5        (rewrite)
```

Comment:
Should you add the seq number of the data files here? maybe have //rewrite as a comment?

Reply:
updated